src/components/tools/table-tool/table-tool.tsx

```diff
@@ -1,85 +1,114 @@
+import { useColumnWidths } from "./use-column-widths";
+import { useContentChangeHandlers } from "./use-content-change-handlers";
 import { useDataSet } from "./use-data-set";
 import { useExpressionsDialog } from "./use-expressions-dialog";
+import { useGeometryLinking } from "./use-geometry-linking";
 import { useGridContext } from "./use-grid-context";
+import { useModelDataSet } from "./use-model-data-set";
 import { useRowLabelColumn } from "./use-row-label-column";
 import { useTableTitle } from "./use-table-title";
+import { useToolApi } from "./use-tool-api";
 import { useCurrent } from "../../../hooks/use-current";
+import { useMeasureText } from "../hooks/use-measure-text";
 import { useToolbarToolApi } from "../hooks/use-toolbar-tool-api";
 
 import "react-data-grid/dist/react-data-grid.css";
 import "./table-tool.scss";
 
 const TableToolComponent: React.FC<IToolTileProps> = ({
-  documentContent, toolTile, model, readOnly,
-  onRequestRowHeight, onRequestUniqueTitle, onRegisterToolApi, onUnregisterToolApi
+  documentId, documentContent, toolTile, model, readOnly, height,
+  onRequestRowHeight, onRequestTilesOfType, onRequestUniqueTitle, onRegisterToolApi, onUnregisterToolApi
 }) => {
+  const modelRef = useCurrent(model);
+  const metadata = (model.content as TableContentModelType).metadata;
+
   const {
     dataSet, columnChanges, triggerColumnChange, rowChanges, triggerRowChange, ...gridModelProps
   } = useModelDataSet(model);
 
+  const handleRequestUniqueTitle = useCallback(() => {
+    return onRequestUniqueTitle(modelRef.current.id);
+  }, [modelRef, onRequestUniqueTitle]);
+
+  const heightRef = useCurrent(height);
+  const handleRequestRowHeight = useCallback((options: { height?: number, deltaHeight?: number }) => {
+    // increase row height automatically but require manual shrinking
+    if (!heightRef.current ||
+        (options?.height && (options.height > heightRef.current)) ||
+        (options?.deltaHeight && (options.deltaHeight > 0))) {
+      onRequestRowHeight(modelRef.current.id, options?.height, options?.deltaHeight);
+    }
+  }, [heightRef, modelRef, onRequestRowHeight]);
+
+  const changeHandlers = useContentChangeHandlers({
+    model, dataSet: dataSet.current,
+    onRequestRowHeight: handleRequestRowHeight, triggerColumnChange, triggerRowChange
+  });
+  const { onSetTableTitle, onSetColumnExpressions, onLinkGeometryTile } = changeHandlers;
+
   const [showRowLabels, setShowRowLabels] = useState(false);
   const {
     ref: gridRef, gridContext, inputRowId, selectedCell, ...gridProps
   } = useGridContext(showRowLabels, triggerColumnChange);
-  const { getTitle, getTitleWidthFromColumns, onBeginTitleEdit, onEndTitleEdit } = useTableTitle({
-    gridContext, model, dataSet: dataSet.current, readOnly, onRequestUniqueTitle: () => onRequestUniqueTitle(model.id)
+  const measureHeaderText = useMeasureText("bold 14px Lato, sans-serif");
+  // const measureBodyText = useMeasureText("14px Lato, sans-serif");
```

Comment:
Is this obsolete or for future use?

Reply:
Placeholder for future use when body cells facto into the column width calculation.

Reply:
Placeholder for when we include body cell contents in column width computations.